### PR TITLE
Make  AM and TaskExecutor runtime interface separate

### PR DESCRIPTION
**Pull request by zuston**
In current framework runtime abstraction, task executor's interface and am's interface are mixed together, which will cause programming troubles and abusing the class variables.
In this PR, let the task executor and am interface seperate.

### CODE REVIEW

**Comment by oliverhu** [comment]
makes sense.. would be good if you can upload a workflow diagram after this change

**Comment by zuston** [comment]
Rebase it @oliverhu

File: tony-core/src/main/java/com/linkedin/tony/Framework.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+package com.linkedin.tony;
+
+import java.io.IOException;
+
+import org.apache.hadoop.conf.Configuration;
+
+import com.linkedin.tony.tensorflow.TonySession;
+import com.linkedin.tony.util.Utils;
+
+public class Framework {
```
Comment (oliverhu) [inline tony-core/src/main/java/com/linkedin/tony/Framework.java:25]: add documentation for this class. `Framework` is a generic name, I don't know a better name either, so better add more documentation explaining

**Review by oliverhu — APPROVED** [review]

File: tony-core/src/main/java/com/linkedin/tony/Framework.java (hunk 1 above)
Comment (zuston) [reply tony-core/src/main/java/com/linkedin/tony/Framework.java:25]: Got it.

**Pull request merged**